site/src/pages/TemplateSettingsPage/TemplateSchedulePage/TemplateSchedulePage.test.tsx

```diff
@@ -89,16 +93,27 @@ const fillAndSubmitForm = async ({
     );
   }
 
-  const submitButton = await screen.findByText(
-    FooterFormLanguage.defaultSubmitLabel,
-  );
+  const submitButton = screen.getByRole("button", {
+    name: FooterFormLanguage.defaultSubmitLabel,
+  });
+
+  expect(submitButton).not.toBeDisabled();
   await user.click(submitButton);
 
-  // User needs to confirm dormancy and autodeletion fields.
+  // User needs to confirm dormancy and auto-deletion fields.
   const confirmButton = await screen.findByTestId("confirm-button");
   await user.click(confirmButton);
 };
 
+function waitForWithCutoff(callback: () => void | Promise<void>) {
```

Comment:
I'm not sure if I get the idea behind `waitForWithCutoff`. In your setup it could be 13sec, but for somebody else, it might be 17sec. I mean, we shouldn't create more time-dependent tests unless there is no way to prevent an event.

Reply:
Oh, that's fair. I keep forgetting that I'm on a Macbook, so the performance is going to be a lot faster. I also agree with you – ideally we shouldn't be introducing time-dependent tests unnecessarily

My worry is that with my understanding of the base `waitFor`, it doesn't fail without some kind of configuration object. As in, it will try to make its assertion periodically, but if the assertion fails, it will just go back to waiting – that failed assertion will just get caught and discarded. So if the assertion doesn't succeed before Jest's global cutoff time runs out, we just get the generic timeout message, instead of something more specific. With some kind of timeout in place, `waitFor` will eventually stop swallowing the failed assertion and throw it to the test suite

I have two other ideas, though
1. Export the global timeout value as a constant, and define the tests to have a hard cutoff of half a second before time is up
2. Just try extending timeouts for the specific tests by passing a custom timeout value to the `describe` functions

I'd be leaning more towards number 1, but they both feel varying degrees of hack-y still

Reply:
Maybe I'm just biased as we have dealt a lot with increasing timeouts endlessly on the backend side to make tests pass. Usually, the clue was to refactor or split the test into 2 separate paths.

I will leave the decision to you folks! cc @BrunoQuaresma

Reply:
Unfortunately, there is no good answer for this 😞 since the test environment for FE tests can be "not very reliable".